subt/tools/ignstate.py

```diff
@@ -0,0 +1,170 @@
+""""
+  Extract robot positions from Ignition SQLite log file
+"""
+import cv2
+import numpy as np
+import re
+import sqlite3
+import xml.etree.ElementTree as ET
+
+from datetime import timedelta
+
+from ignition.msgs.pose_v_pb2 import Pose_V
+
+COLORS = [
+    (255, 255, 255),
+    (0, 255, 0),
+    (0, 0, 255),
+    (255, 0, 0),
+    (0, 255, 255),
+    (255, 0, 255),
+    (255, 255, 0),
+]
+SCALE = 10  # 1 pixel is 1dm
+BORDER_PX = 10  # extra border
+
+MARKERS = {
+    'backpack': (cv2.MARKER_SQUARE, 3),
+    'rescue_randy': (cv2.MARKER_DIAMOND, 2),
+    'gas': (cv2.MARKER_TRIANGLE_UP, 1),
+    'vent': (cv2.MARKER_STAR, 4),
+    'phone': (cv2.MARKER_TILTED_CROSS, 5),
+    'artifact_origin': (cv2.MARKER_CROSS, 6),
+}
+
+
+def read_poses(filename, seconds=3700):
```

Comment:
why not 3600s?

Reply:
That is a simulated time and the scoring does not start exactly at time equal 0 but some moments later. The goal here is to get all poses until the scoring has been finished.